### ISSUE TITLE
Add a static site buildpack, useful for react and vue apps, can be nginx or other web server

### ISSUE THREAD

**Issue by yairyairyair** [GoogleCloudPlatform/buildpacks#309]
**Is your feature request related to a problem? Please describe.**
If this feature is related to an existing issue, please provide a link to that as well

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

It can be detected via if has folder build or dist,
if yes then should add nginx with SPA/pushstate config
Seems useful because frontend static apps currently are hard to run in google buildpacks

**Comment by yairyairyair** [comment]
Also there is already an nginx buildpack as utils.

And the detect script can also detect an nginx.conf and apply it instead for further customization

**Comment by jama22** [comment]
I'm assuming that you're building your react/vue apps in CI/CD pipeline, and you just want to deploy the compiled files using a static site buildpack?

**Comment by yairyairyair** [comment]
There is no CI CD, 
Basically, The pack build command runs npm run build by default as it should, and then runs npm start.
It should in my opinion check via some env param if it is an SPA frontend and if so then instead of npm start should serve the dist/build folder

**Comment by jama22** [comment]
Ah okay, that makes a lot of sense. We've been seeing this kind of use case/request a lot. It could similarly help with easily serving markdown-generated HTML sites  (e.g. Hugo, Jekyll), or pre-compiled applications (e.g. Go executables, Java, Java native images, etc.)

**Comment by yairyairyair** [comment]
Yes indeed could be useful, will this issue be in like a roadmap or something?
I would have tried coding it yet have no knowledge about buildpacks

**Comment by jvkassi** [comment]
+1
